system/src/system_cloud.cpp

```diff
@@ -38,8 +44,36 @@
 #include "deviceid_hal.h"
 #include "system_mode.h"
 
+#if PLATFORM_THREADING
+#include "spark_wiring_timer.h"
+#endif // PLATFORM_THREADING
+
 extern void (*random_seed_from_cloud_handler)(unsigned int);
 
+namespace {
+
+using namespace particle::system;
+
+#if PLATFORM_THREADING
+  Timer _vitals_timer(std::numeric_limits<unsigned>::max(), []() -> void {
+      const auto task = new(std::nothrow) ISRTaskQueue::Task;
+      if (!task) {
+          return;
+      }
+      task->func = [](ISRTaskQueue::Task* task) {
+          delete task;
+          spark_protocol_post_description(spark_protocol_instance(), particle::protocol::DESCRIBE_METRICS, nullptr);
+      };
+      SystemISRTaskQueue.enqueue(task);
+  }, false);
+  VitalsPublisher<Timer> _vitals(std::bind(spark_protocol_post_description, spark_protocol_instance(), particle::protocol::DESCRIBE_METRICS, nullptr), &_vitals_timer);
```

Comment:
I suggest to use a C++ singleton here:
```cpp
VitalsPublisher<Timer>* vitalsPublisher() {
    static VitalsPublisher<Timer> vitals(...);
    return &vitals;
}
```
That would make the initialization independent of the order in which globals are initialized across multiple translation units in C++. As a bonus, the resources, such as a timer instance, will be allocated on demand, i.e. only if the application does publish device vitals.